src/OmniSharp.Roslyn/WorkspaceServices/ExtractInterfaceWorkspaceService.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Castle.DynamicProxy;
+using Microsoft.CodeAnalysis;
+
+namespace OmniSharp
+{
+    public class ExtractInterfaceWorkspaceService : IInterceptor
+    {
+        public void Intercept(IInvocation invocation)
```

Comment:
To avoid a dependency on Castle.Core, you can have a look in [MetadataHelper](https://github.com/OmniSharp/omnisharp-roslyn/blob/c9d45c06a58d33e02c16328afb2a08f8f3db5dfe/src/OmniSharp.Roslyn/MetadataHelper.cs#L66) on how this could be done.